lib/brakeman/report/report_sarif.rb

```diff
@@ -0,0 +1,89 @@
+require 'brakeman/report/report_json'
+
+class Brakeman::Report::SARIF < Brakeman::Report::JSON
+  def generate_report
+    sarif_log = {
+      :version => '2.1.0',
+      :$schema => 'https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.4.json',
+      :runs => runs,
+    }
+    JSON.pretty_generate sarif_log
+  end
+
+  def runs
+    [
+      {
+        :tool => {
+          :driver => {
+            :name => 'Brakeman',
+            :informationUri => 'https://brakemanscanner.org',
+            :rules => rules,
+          },
+        },
+        :artifacts => artifacts,
+        :results => results,
+      },
+    ]
+  end
+
+  def rules
+    @rules ||= unique_warnings.map do |warning|
+      check_name = warning.check.gsub(/^Brakeman::Check/, '')
+      check_description = check_descriptions[check_name]
+      {
+        :id => warning.warning_code.to_s,
+        :shortDescription => {
+          :text => check_description,
```

Comment:
:wave: @presidentbeef please let me know if there's a better way to get this description. You mentioned elsewhere:

> Just as a note: checks/rules in Brakeman can emit more than one 
warning_code.

And I think I see what you mean now. As an example, rules "12" and "77" are both emitted by the same `DefaultRoute` check, and so they share the same description. That might be OK, but I am not sure. Is there a way to get a description for each warning_code emitted by a check?

Reply:
Unfortunately, that's the best we have right now. :(